File: 20221102/모범답안/20221102_07/beautiful.js
/**
 * @param {array} beautiful
 * @return {int}
 */
function solution(beautiful) {
    let ans = Infinity;
    let n = beautiful.length;

    for (let i = 0; i < n; i++) {
      for (let j = i + 1; j < n; j++) {
        let cur = 0;
        for (let k = 0; k < n; k++) {
          if ((beautiful[j] - beautiful[i]) * (k - i) != (beautiful[k] - beautiful[i]) * (j - i)) cur++;
        }
        ans = Math.min(ans, cur);
      }
    }

    return ans == Infinity ? 0 : ans;
}

File: 20221102/모범답안/20221102_07/beautiful.py
def solution(beautiful):
    '''
    :param beautiful: list
    :return: int
    '''
    n = len(beautiful)

    if n == 1:
        return 0
    else:
        answer = float("inf")
        for j in range(n):
            freq = {}
            for i in range(n):
                if i != j:
                    d = round((beautiful[i] - beautiful[j])/(i-j), 10)
                    freq[d] = freq.get(d, 0) + 1
            answer = min(answer, n - 1 - max(freq.values()))
        
        return answer
